### PR TITLE
Re do gold logic

### PR DESCRIPTION
**Pull request by psakievich**
Make gold logic so that golds can be preserved with a spack install and unify implementation across packages.

Long term goal is to support golds to diff against the last release on system level installs.
Diffs could then be used to steer the release cycle.

### CODE REVIEW

**Review by marchdf — APPROVED** [review]
Awesome work!

**Comment by psakievich** [comment]
`deploy.py` is broken on `spack@develop` see https://github.com/spack/spack/issues/43555.  Need to resolve this before merging this PR.

File: repos/exawind/packages/nalu-wind/package.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Copyright (c) 2022, National Technology & Engineering Solutions of Sandia,
-# LLC (NTESS). Under the terms of Contract DE-NA0003525 with NTESS, the U.S.
+
```
Comment (jrood-nrel) [inline repos/exawind/packages/nalu-wind/package.py:2]: I think this is a mistake.

File: repos/exawind/packages/trilinos/package.py [hunk 2]
```diff
@@ -9,15 +9,15 @@
...
 find_machine = importlib.import_module("find-exawind-manager")
 
-class Trilinos(bTrilinos, CmakeExtension):
+class Trilinos(bTrilinos, CtestPackage):
     # Our custom release versions should be the latest release tag found on
     # the trilinos github page appended with the date of the commit.
     # this preserves the Trilinos versioning scheme and will allow for valid
     # version comparisons in spack's internals.
-    version("14.1.0.2023.02.28", commit="8b3e2e1db4c7e07db13225c73057230c4814706f")
+    version("13.4.1.2023.02.28", commit="8b3e2e1db4c7e07db13225c73057230c4814706f")
```
Comment (jrood-nrel) [inline repos/exawind/packages/trilinos/package.py:20]: I think this requires changes in machine configs too.

Comment (psakievich) [reply repos/exawind/packages/trilinos/package.py:20]: Yeah you're right.

**Review by jrood-nrel — APPROVED** [review]

**Pull request merged**